modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/servlet/taglib/JSImportMapsExtenderTopHeadDynamicInclude.java

```diff
@@ -96,53 +99,32 @@ public void register(DynamicIncludeRegistry dynamicIncludeRegistry) {
 		dynamicIncludeRegistry.register("/html/common/themes/top_head.jsp#pre");
 	}
 
-	public JSImportMapsRegistration register(
-		String scope, JSONObject jsonObject) {
-
-		if (scope == null) {
-			long globalId = _nextGlobalId.getAndIncrement();
-
-			_globalImportMapJSONObjects.put(globalId, jsonObject);
-
-			_rebuildImportMaps();
-
-			return new JSImportMapsRegistration() {
-
-				@Override
-				public void unregister() {
-					_globalImportMapJSONObjects.remove(globalId);
-				}
-
-			};
-		}
-
-		_scopedImportMapJSONObjects.put(scope, jsonObject);
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_bundleContext = bundleContext;
 
 		_rebuildImportMaps();
 
-		return new JSImportMapsRegistration() {
+		_serviceTracker = new ServiceTracker<>(
+			bundleContext, JSImportMapsContributor.class,
+			_serviceTrackerCustomizer);
 
-			@Override
-			public void unregister() {
-				_scopedImportMapJSONObjects.remove(scope);
-			}
+		_serviceTracker.open();
 
-		};
+		modified();
```

Comment:
Hey @MrJohn1911, put the `modified` right after the `_bundleContext` initialization.